lib/sisimai/bite/email/facebook.rb

```diff
@@ -0,0 +1,274 @@
+                  # Diagnostic-Code: smtp; 550 5.1.1 RCP-P2
+                  #     http://postmaster.facebook.com/response_codes?ip=192.0.2.135#rcp Refused due to recipient preferences
+                  v['spec'] = cv[1].upcase
+                  v['diagnosis'] = cv[2]
+
+                elsif p =~ /\A[Dd]iagnostic-[Cc]ode:[ ]*/ && cv = e.match(/\A[ \t]+(.+)\z/)
+                  # Continued line of the value of Diagnostic-Code header
+                  v['diagnosis'] ||= ''
+                  v['diagnosis']  += ' ' + cv[1]
+                  havepassed[-1] = 'Diagnostic-Code: ' + e
+                end
+              end
+
+            else
+              if cv = e.match(/\A[Rr]eporting-MTA:[ ]*(?:DNS|dns);[ ]*(.+)\z/)
+                # Reporting-MTA: dns; mx.example.jp
+                next if connheader['lhost'].size > 0
+                connheader['lhost'] = cv[1].downcase
+                connvalues += 1
+
+              elsif cv = e.match(/\A[Aa]rrival-[Dd]ate:[ ]*(.+)\z/)
+                # Arrival-Date: Wed, 29 Apr 2009 16:03:18 +0900
+                next if connheader['date'].size > 0
+                connheader['date'] = cv[1]
+                connvalues += 1
+              end
+
+            end
+          end
+        end
+        return nil if recipients.zero?
+        require 'sisimai/string'
+
+        dscontents.map do |e|
+          e['lhost']   ||= connheader['lhost']
+          e['agent']     = self.smtpagent
+          e['diagnosis'] = Sisimai::String.sweep(e['diagnosis'])
+
+          if cv = e['diagnosis'].match(/\b([A-Z]{3})[-]([A-Z])(\d)\b/)
+            # Diagnostic-Code: smtp; 550 5.1.1 RCP-P2
+            lhs = cv[1]
+            rhs = cv[2]
+            num = cv[3]
+
+            fbresponse = sprintf('%s-%s%d', lhs, rhs, num)
+          end
+
+          catch :SESSION do
+            ReFailure.each_key do |r|
+              # Verify each regular expression of session errors
+              ReFailure[r].each do |rr|
+                # Check each regular expression
+                next unless fbresponse == rr
+                e['reason'] = r.to_s
+                throw :SESSION
+              end
+            end
+          end
+
+          unless e['reason']
+            # http://postmaster.facebook.com/response_codes
+            #   Facebook System Resource Issues
+            #   These codes indicate a temporary issue internal to Facebook's
+            #   system. Administrators observing these issues are not required to
+            #   take any action to correct them.
+            if fbresponse =~ /\AINT-T\d+\z/
+              # * INT-Tx
+              #
+              # https://groups.google.com/forum/#!topic/cdmix/eXfi4ddgYLQ
+              # This block has not been tested because we have no email sample
+              # including "INT-T?" error code.
+              e['reason'] = 'systemerror'
+            end
+          end
+        end
+
+        rfc822part = Sisimai::RFC5322.weedout(rfc822list)
```

Comment:
__[rubocop]__
* Style/Next: Use `next` to skip iteration. [:ref](https://github.com/bbatsov/ruby-style-guide#no-nested-conditionals)

<!-- Comment by [SideCI](https://sideci.com) -->